icu4c/source/test/cintltst/ccaltst.c

```diff
@@ -2826,4 +2827,45 @@ TestFWWithISO8601() {
     }
 }
 
+void
+TestGetIanaTimeZoneID() {
+    const UChar* UNKNOWN = u"Etc/Unknown";
+    typedef struct {
+        const UChar* id;
+        const UChar* expected;
+    } IanaTimeZoneIDTestData;
+    
+    const IanaTimeZoneIDTestData TESTDATA[] = {
+        {u"",                   UNKNOWN},
+        {0,                     UNKNOWN},
+        {u"America/New_York",   u"America/New_York"},
+        {u"Asia/Calcutta",      u"Asia/Kolkata"},
+        {u"Europe/Kiev",        u"Europe/Kyiv"},
+        {u"Europe/Zaporozhye",  u"Europe/Kyiv"},
+        {u"Etc/GMT-1",          u"Etc/GMT-1"},
+        {u"Etc/GMT+20",         UNKNOWN},
+        {u"PST8PDT",            u"PST8PDT"},
+        {u"GMT-08:00",          UNKNOWN},
+        {0,                     0}
```

Comment:
we should aslo test the case which the id is UNKNOWN and make sure the return of that is correct.

Reply:
Hmm.. it looks I missed the case. Will do.

Reply:
done